.github/workflows/building.yml

```diff
@@ -0,0 +1,129 @@
+name: Building Wheels
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+
+  wheel:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-18.04]
+        python-version: ['3.7', '3.8', '3.9', '3.10']
+        torch-version: [1.9.0, 1.10.0, 1.11.0, 1.12.0]
+        cuda-version: ['cpu', 'cu102', 'cu111', 'cu113', 'cu115', 'cu116']
+        exclude:
+          - torch-version: 1.9.0
+            cuda-version: 'cu113'
+          - torch-version: 1.9.0
+            cuda-version: 'cu115'
+          - torch-version: 1.9.0
+            cuda-version: 'cu116'
+          - torch-version: 1.9.0
+            python-version: '3.10'
+          - torch-version: 1.10.0
+            cuda-version: 'cu115'
+          - torch-version: 1.10.0
+            cuda-version: 'cu116'
+          - torch-version: 1.10.0
+            python-version: '3.10'
+          - torch-version: 1.11.0
+            cuda-version: 'cu111'
+          - torch-version: 1.11.0
+            cuda-version: 'cu116'
+          - torch-version: 1.12.0
+            cuda-version: 'cu111'
+          - torch-version: 1.12.0
+            cuda-version: 'cu115'
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Free up disk space
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          sudo rm -rf /usr/share/dotnet
+
+      - name: Install ninja
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install -y ninja-build
+
+      - name: Install CUDA ${{ matrix.cuda-version }}
+        if: ${{ matrix.cuda-version != 'cpu' }}
+        run: |
+          bash .github/workflows/cuda/${{ matrix.cuda-version }}-${{ runner.os }}.sh
```

Comment:
Could we use pre-built containers to be more scalable? You may refer to the usage in MMCV (https://github.com/open-mmlab/mmcv/blob/master/.github/workflows/build.yml#L152-L153). Please also check all the supported versions at https://hub.docker.com/r/pytorch/pytorch/tags to see whether it is comprehensive enough.